Disallow stored scripts from runtime field contexts

jdconrad opened the pull request:
This change disallows stored scripts using a runtime field context. This adds a constructor parameter to script context as to whether or not a context is allowed to be a stored script.

jdconrad committed:
change context names for run time fields to type_field

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

jdconrad committed:
pare script constructors to 2

nik9000 reviewed (commented):
I think it'd be nice to add a rest test for this just to make sure its all plugged in on all the layers. I might not even make the change to `StoredScriptsIT` because the transport client is deprecated.

jdconrad committed:
add yaml tests

jdconrad commented:
@nik9000 Removed StoredScriptIT tests and replaced them with yml tests instead. I added the yml tests to the Painless module as I wasn't really sure what the best place for them would be. Open to suggestions (@rjernst).

rjernst commented:
> I added the yml tests to the Painless module as I wasn't really sure what the best place for them would be

The painless module is the only place for them at the moment, since the yaml tests in core run without any modules.

nik9000 commented:
I think the painless module is the most reasonable place we have for them at the moment, yeah.

nik9000 reviewed (approved):
LGTM
